sycl/include/CL/sycl/pipes.hpp

```diff
@@ -0,0 +1,89 @@
+//==---------------- pipes.hpp - SYCL pipes ------------*- C++ -*-----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/__spirv/spirv_types.hpp>
+#include <CL/sycl/stl.hpp>
+
+namespace cl {
+namespace sycl {
+
+template <class name, class dataT, int32_t min_capacity = 0> class pipe {
+public:
+  // Non-blocking pipes
+  // Reading from pipe is lowered to SPIR-V instruction OpReadPipe via SPIR-V
+  // friendly LLVM IR.
+  static dataT read(bool &Success) {
+#ifdef __SYCL_DEVICE_ONLY__
+    RPipeTy<dataT> RPipe =
+      __spirv_CreatePipeFromPipeStorage_read<dataT>(&m_Storage);
+    dataT TempData;
```

Comment:
So `dataT` has to have a default constructor, a copy constructor and a destructor. Are we OK with this?

Reply:
Yes.

Reply:
Then this should to be reflected in the documentation.

Reply:
Added